cmd/metal-api/internal/service/firewall-service.go

```diff
@@ -215,6 +231,42 @@ func (r *firewallResource) allocateFirewall(request *restful.Request, response *
 	r.send(request, response, http.StatusOK, resp)
 }
 
+func (r firewallResource) setVPNConfigInSpec(allocationSpec *machineAllocationSpec) error {
+	if r.headscaleClient == nil {
+		return nil
+	}
+
+	ctx := context.Background()
+
+	// Get project VPN namespace
+	projectID := allocationSpec.ProjectID
+	p, err := r.mdc.Project().Get(ctx, &mdmv1.ProjectGetRequest{Id: projectID})
+	if err != nil {
+		return err
+	}
+	if p.GetProject() == nil {
+		return fmt.Errorf("Project with ID %s wasn't found", projectID)
+	}
+
+	// Try to create namespace in Headscale DB
+	if err = r.headscaleClient.CreateNamespace(p.Project.Name); err != nil {
+		return fmt.Errorf("failed to create new VPN namespace for the Project: %w", err)
+	}
+
+	expiration := time.Now().Add(90 * 24 * time.Hour)
```

Comment:
If we do not need the rotation and it's really just a join token that's being requested from the headscale server, we can reduce the expiration time to two hours or something (something small).

Reply:
Unfortunately we need the rotation. Tested connection to headscale after token is expired(initial connection with non-expired token) -- it stops working after that.